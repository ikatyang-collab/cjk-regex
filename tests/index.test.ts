import * as cjk from '../src/index.js'
import { test, expect } from 'vitest'

const testCases: {
  [char: string]: 'non-cjk' | 'cjk-letter' | 'cjk-punctuation'
} = {
  '.': 'non-cjk',
  'a': 'non-cjk',
  '。': 'cjk-punctuation',
  '⿱': 'cjk-punctuation',
  '中': 'cjk-letter',
  'ㄅ': 'cjk-letter',
  '𬉼': 'cjk-letter',
  'あ': 'cjk-letter',
  'ㅂ': 'cjk-letter',
  '가': 'cjk-letter',
  'ퟔ': 'cjk-letter',
  '〤': 'cjk-letter',
  '𛀂': 'cjk-letter',
  'ｦ': 'cjk-letter',
  '々': 'cjk-letter',
}

Object.keys(testCases).forEach(character => {
  const category = testCases[character]
  const title = `"${character}" (0x${character
    .charCodeAt(0)
    .toString(16)}) is ${category}`
  test(title, () => {
    switch (category) {
      case 'non-cjk':
        expect(character).not.toMatch(cjk.all().toRegExp())
        expect(character).not.toMatch(cjk.letters().toRegExp())
        expect(character).not.toMatch(cjk.punctuations().toRegExp())
        break
      case 'cjk-letter':
        expect(character).toMatch(cjk.all().toRegExp())
        expect(character).toMatch(cjk.letters().toRegExp())
        expect(character).not.toMatch(cjk.punctuations().toRegExp())
        break
      case 'cjk-punctuation':
        expect(character).toMatch(cjk.all().toRegExp())
        expect(character).not.toMatch(cjk.letters().toRegExp())
        expect(character).toMatch(cjk.punctuations().toRegExp())
        break
      default:
        throw new Error(`Unexpected category "${category}"`)
    }
  })
})
