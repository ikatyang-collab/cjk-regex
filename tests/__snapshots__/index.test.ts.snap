// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot: all 1`] = `
[
  [
    4352,
    4607,
  ],
  [
    11904,
    11929,
  ],
  [
    11931,
    12019,
  ],
  [
    12032,
    12245,
  ],
  [
    12272,
    12351,
  ],
  [
    12353,
    12438,
  ],
  [
    12445,
    12447,
  ],
  [
    12449,
    12538,
  ],
  [
    12541,
    12543,
  ],
  [
    12549,
    12591,
  ],
  [
    12593,
    12686,
  ],
  [
    12688,
    19903,
  ],
  [
    19968,
    40959,
  ],
  [
    43360,
    43388,
  ],
  [
    44032,
    55203,
  ],
  [
    55216,
    55238,
  ],
  [
    55243,
    55291,
  ],
  [
    63744,
    64109,
  ],
  [
    64112,
    64217,
  ],
  [
    65040,
    65055,
  ],
  [
    65072,
    65135,
  ],
  [
    65280,
    65519,
  ],
  [
    94179,
    94179,
  ],
  [
    110576,
    110579,
  ],
  [
    110581,
    110587,
  ],
  [
    110589,
    110590,
  ],
  [
    110592,
    110882,
  ],
  [
    110898,
    110898,
  ],
  [
    110928,
    110930,
  ],
  [
    110933,
    110933,
  ],
  [
    110948,
    110951,
  ],
  [
    127488,
    127488,
  ],
  [
    131072,
    173791,
  ],
  [
    173824,
    177977,
  ],
  [
    177984,
    178205,
  ],
  [
    178208,
    183969,
  ],
  [
    183984,
    191456,
  ],
  [
    191472,
    192093,
  ],
  [
    194560,
    195101,
  ],
  [
    196608,
    201546,
  ],
  [
    201552,
    205743,
  ],
]
`;

exports[`Snapshot: letters 1`] = `
[
  [
    4352,
    4607,
  ],
  [
    11904,
    11929,
  ],
  [
    11931,
    12019,
  ],
  [
    12032,
    12245,
  ],
  [
    12293,
    12293,
  ],
  [
    12295,
    12295,
  ],
  [
    12321,
    12329,
  ],
  [
    12344,
    12347,
  ],
  [
    12353,
    12438,
  ],
  [
    12445,
    12447,
  ],
  [
    12449,
    12538,
  ],
  [
    12541,
    12543,
  ],
  [
    12549,
    12591,
  ],
  [
    12593,
    12686,
  ],
  [
    12704,
    12735,
  ],
  [
    12784,
    12830,
  ],
  [
    12896,
    12926,
  ],
  [
    13008,
    13054,
  ],
  [
    13056,
    13143,
  ],
  [
    13312,
    19903,
  ],
  [
    19968,
    40959,
  ],
  [
    43360,
    43388,
  ],
  [
    44032,
    55203,
  ],
  [
    55216,
    55238,
  ],
  [
    55243,
    55291,
  ],
  [
    63744,
    64109,
  ],
  [
    64112,
    64217,
  ],
  [
    65382,
    65391,
  ],
  [
    65393,
    65437,
  ],
  [
    65440,
    65470,
  ],
  [
    65474,
    65479,
  ],
  [
    65482,
    65487,
  ],
  [
    65490,
    65495,
  ],
  [
    65498,
    65500,
  ],
  [
    94179,
    94179,
  ],
  [
    110576,
    110579,
  ],
  [
    110581,
    110587,
  ],
  [
    110589,
    110590,
  ],
  [
    110592,
    110882,
  ],
  [
    110898,
    110898,
  ],
  [
    110928,
    110930,
  ],
  [
    110933,
    110933,
  ],
  [
    110948,
    110951,
  ],
  [
    127488,
    127488,
  ],
  [
    131072,
    173791,
  ],
  [
    173824,
    177977,
  ],
  [
    177984,
    178205,
  ],
  [
    178208,
    183969,
  ],
  [
    183984,
    191456,
  ],
  [
    191472,
    192093,
  ],
  [
    194560,
    195101,
  ],
  [
    196608,
    201546,
  ],
  [
    201552,
    205743,
  ],
]
`;

exports[`Snapshot: punctuations 1`] = `
[
  [
    12272,
    12292,
  ],
  [
    12294,
    12294,
  ],
  [
    12296,
    12320,
  ],
  [
    12330,
    12343,
  ],
  [
    12348,
    12351,
  ],
  [
    12688,
    12703,
  ],
  [
    12736,
    12783,
  ],
  [
    12831,
    12895,
  ],
  [
    12927,
    13007,
  ],
  [
    13055,
    13055,
  ],
  [
    13144,
    13311,
  ],
  [
    65040,
    65055,
  ],
  [
    65072,
    65135,
  ],
  [
    65280,
    65381,
  ],
  [
    65392,
    65392,
  ],
  [
    65438,
    65439,
  ],
  [
    65471,
    65473,
  ],
  [
    65480,
    65481,
  ],
  [
    65488,
    65489,
  ],
  [
    65496,
    65497,
  ],
  [
    65501,
    65519,
  ],
]
`;
